build-system/eslint-rules/jsx-namespace.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+
+const defaultNamespace = {
+  // We expect the compiler to output to this namespace by default.
+  // As such, we insert a @jsx annotation only when using different names.
+  name: 'Preact',
+  from: '#core/dom/preact',
+};
+
+// Enforces importing a namespace specifier if using JSX
+//
+// Good
+// import * as Preact from '#preact';
+// <div />
+//
+// Bad
+// <div />
+//
+// Bad
+// import { createElement } from '#preact';
+// <div />
+module.exports = {
+  meta: {
+    fixable: 'code',
+  },
+
+  create(context) {
+    /**
+     * @param {*} node
+     */
+    function requireJsxNamespace(node) {
+      if (imported) {
+        return;
+      }
+      if (warned) {
+        return;
+      }
+      warned = true;
+
+      const [{from, name} = defaultNamespace] = context.options;
+
+      const comment =
+        name !== defaultNamespace.name
+          ? `/** @jsx ${name}.createElement */\n`
+          : '';
+      const decl = `import * as ${name} from '${from}';\n`;
+
+      const inserted = comment + decl;
+
+      context.report({
+        node,
+        message: [
+          `Using JSX requires importing the '${name} namespace, eg:`,
+          '```',
+          inserted.trim(),
+          '```',
+        ].join('\n\t'),
+
+        fix(fixer) {
+          const ancestors = context.getAncestors();
+          const program = ancestors[0];
+          let firstImport = program.body.find(
+            (node) => node.type === 'ImportDeclaration'
+          );
+          if (!firstImport) {
+            firstImport = ancestors[1];
+          }
+
+          return fixer.insertTextBefore(firstImport, inserted);
+        },
+      });
+    }
+
+    let imported = false;
+    let warned = false;
+
+    return {
+      Program(node) {
+        const hasAnnotation = node.comments.find(
+          ({type, value}) => type === 'Block' && value.includes('* @jsx ')
+        );
+        imported = hasAnnotation;
```

Comment:
Shouldn't this be a separate variable? As is, just having a `/* @jsx Foo.createElement */` will allow you to skip the verification that you've imported `import * as Foo from '...'`